pincer/core/gateway.py

```diff
@@ -46,275 +77,360 @@ class Dispatcher:
     """
 
     def __init__(
-            self, token: str, *,
-            handlers: Dict[int, Handler],
+            self,
+            token: str, *,
             intents: Intents,
-            reconnect: bool
+            url: str,
+            shard: int,
+            num_shards: int
     ) -> None:
         if len(token) != 59:
             raise InvalidTokenError(
                 "Discord Token must have exactly 59 characters."
             )
 
-        self.__token = token
-        self.__keep_alive = True
-        self.__has_closed = self.__should_restart = False
-        self.__socket: Optional[WebSocketClientProtocol] = None
-        self.__intents = intents
-        self.__reconnect = reconnect
-
-        async def identify_and_handle_hello(
-                socket: WebSocketClientProtocol,
-                payload: GatewayDispatch
-        ):
-            """|coro|
-
-            Identifies the client to the Discord Websocket API, this
-            gets done when the client receives the ``hello`` (opcode 10)
-            message from discord. Right after we send our identification
-            the heartbeat starts.
-
-            Parameters
-            ----------
-            socket : :class:`~ws:websockets.legacy.client.WebSocketClientProtocol`
-                The current socket, which can be used to interact
-                with the Discord API.
-            payload : :class:`~pincer.core.dispatch.GatewayDispatch`
-                The received payload from Discord.
```

Comment:
I refactored so `payload` was the only parameter. At that point I think adding a docstring does not add any clarity because its an internal method.

Reply:
i just realized i did write a docstring for that buts its much shorter. I didn't put a class docstring because comment annotations are so much easier to look at since theres less scrolling. I don't think anyone is going to be looking at the API docs that closely for the gateway cause you usually look at code when working on internals lol.

Reply:
I ended up adding more docs anyway